### PR TITLE
Business logic layer's datamodel integrateion (logic side) resolves #13

### LINKED ISSUE #13: Integrate Business logic layer's datamodel with the data contract on the WCF side
Currently the 2 differ heavily

### PR DESCRIPTION
**Pull request by matyimat**

### CODE REVIEW

File: Dixit_Logic/Interfaces/IDixitGame.cs [hunk 1]
```diff
@@ -12,58 +12,23 @@ namespace Dixit_Logic.Interfaces
...
-            set;
-        }
-
-        /// <summary>
-        /// This add a new player (by name) to the Players.
+        /// This add a new player (by name) to the ActualGameState.
         /// </summary>
         /// <param name="name">The name of the player.</param>
         /// <returns>The new player identifier</returns>
         int AddPlayer(String name);
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDixitGame.cs]: How about instead of returning a simple int, return an IPlayer here?

Comment (VSZM) [reply Dixit_Logic/Interfaces/IDixitGame.cs]: Also in the PutCard method.. It is not an int that is putting the card, but a Player. Makes the code more readable.

**Review by VSZM — COMMENTED** [review]
Please fix these

File: Dixit_Logic/Interfaces/IDixitGame.cs (hunk 1 above)
Comment (matyimat) [reply Dixit_Logic/Interfaces/IDixitGame.cs]: I ll correct these mistakes after Marci ll resolve conflicts in his pull request to avoid more conflicts.

Comment (VSZM) [reply Dixit_Logic/Interfaces/IDixitGame.cs]: I think you can go ahead, you will not conflict.

File: Dixit_Logic/Interfaces/IGameState.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dixit_Logic.Interfaces
+{
+    /// <summary>
+    /// It describe a state of game. So, it keep every necessary information about a game's moment.
+    /// </summary>
+    interface IGameState
```
Comment (SilverGeri) [inline Dixit_Logic/Interfaces/IGameState.cs]: please make interfaces public

File: Dixit_Logic/Interfaces/IDeck.cs [hunk 3]
```diff
@@ -12,21 +12,26 @@ namespace Dixit_Logic.Interfaces
     interface IDeck
     {
         /// <summary>
-        /// It add optional piece of cards to the deck.
+        /// This contain the deck's cards. 
         /// </summary>
-        /// <param name="cards">Array of cards what you want to add.</param>
-        void AddCards(IList<ICard> cards);
+        IList<ICard> Cards
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDeck.cs:17]: public

File: Dixit_Logic/Interfaces/IDeck.cs [hunk 4]
```diff
@@ -12,21 +12,26 @@ namespace Dixit_Logic.Interfaces
     interface IDeck
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDeck.cs]: public

File: Dixit_Logic/Interfaces/IDeck.cs [hunk 5]
```diff
@@ -12,21 +12,26 @@ namespace Dixit_Logic.Interfaces
...
+            get;
+        }
 
         /// <summary>
-        /// It change the card order in the deck, what make a new drawing order.
+        /// It add optional piece of cards to the deck.
         /// </summary>
-        void Shuffle();
+        /// <param name="cards">Array of cards what you want to add.</param>
+        void AddCards(IList<ICard> cards);
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDeck.cs:26]: public...

File: Dixit_Logic/Interfaces/IDixitGame.cs [hunk 6]
```diff
@@ -12,58 +12,23 @@ namespace Dixit_Logic.Interfaces
     interface IDixitGame
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDixitGame.cs]: public!

File: Dixit_Logic/Interfaces/IDixitGame.cs [hunk 7]
```diff
@@ -77,11 +42,9 @@ IDeck BoardDeck
         void NewGuess(int player, int card);
 
         /// <summary>
-        /// This evaluates the points of actual turn and adds these to the appropriate players's point. 
-        /// In the end, it returns with the result of evaluation (pairs of player id and point).
+        /// This evaluates the points of actual turn and adds these to the appropriate players's point (in ActualGameState).         
         /// </summary>
-        /// <returns>A dictionary, what has a player id "key" with a point "value".</returns>
-        Dictionary<int, int> EvaluatePoints();
+        void EvaluatePoints();
```
Comment (VSZM) [inline Dixit_Logic/Interfaces/IDixitGame.cs:47]: public

**Review by VSZM — CHANGES_REQUESTED** [review]

**Comment by VSZM** [comment]
Matyi in the latest commit you still have not made the interfaces public. Please make all of the interfaces public so I can approve.

**Review by VSZM — APPROVED** [review]

**Pull request merged**